### ISSUE TITLE
Change extensions to contain a single PSExtension Object

### ISSUE THREAD

**Issue by LeonMatthes** [hpi-swa-teaching/SWT18-Project-09#59]
Context menu and acceptsRescale should be handled through a single extension Object, instead of the current approach.
This would allow users to add PowerSqueak context menus to Morphs, without adding the morphs code.
This might also allow us to better solve #56

**Comment by LeonBein** [comment]
I would not agree on that. In my opion the different types of extensions should not be mixed up in order to allow greater flexibility. It does not seem right to allow a morph to have a custom context menu only if it accepts rescales and vice versa.

**Comment by MrModder** [comment]
I agree with the argument, but I would still say, we should use a single extension object, which contains all the information, which features are supported by the extension, and, more importantly, this object should contain blocks to execute, if the feature is supported.
In contrast to the current system, which basically checks, whether a Morph understands a function by checking, whether the Morph has a specific extension and then executes it, this would make for a cleaner API for anyone, who would want to better integrate a previously unsupported Morph into PowerSqueak.
What I imagine such a procedure would look like:
``` smalltalk
Morph>>initialize
    
    | psExtension |
    psExtension := PSExtension new. "by default, all features are disabled"
    psExtension rescale: [ :extent | self rescale: extent ]
    self setProperty: #PSExtension to: psExtension
```

While a bit more bloated, it would allow you, to easily adapt any Morph to work with PowerSqueak, even if you don't want to change its code, and it would not add as much to the extension interface, as we do have to share the extensions with other external tools, which may lead to conflicts down the road.

**Comment by LeonBein** [comment]
Yet, the ability to add a block for the extensions does not require a specific class, as the block may also simply be added as extension:
```smalltalk
XYMorph>>initialize

   super initialize.
   self setProperty: #rescale to: [:newExtent | self rescale: newExtent]
```
Two more points: 
Remember Pheno getting into everything? We should avoid that.
Also, adding extensions with one object per extension makes us much more flexible while still not overengineering stuff. Different extensions should be decoupled from each other.
The only point I see is that some extensions might need more than one method. However, this is currently hypothetic and could also easily be avoided by going back to the "extension = flag" approach. (I think your current approach doesn't solve this problem either)
